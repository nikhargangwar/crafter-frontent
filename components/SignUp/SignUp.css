@import url('https://fonts.googleapis.com/css2?family=Bebas+Neue&display=swap');

.sign-up {
  height: 100vh;
  background-color: #211E1E;
  display: flex;
  align-items: center;
  justify-content: center;
}

.sign-up img {
  position: relative;
  bottom: 30px;
  cursor: pointer;
}

.sign-up-form {
  width: 100%;
  height: 100%;
  justify-content: center;
}

.sign-up-form-container {
  width: 100%;
  align-items: center;
  /* margin-top: 40px; */
}

.sign-header p {
  cursor: pointer;
  color: rgb(65, 78, 172);
  border-bottom: 3px solid rgb(160, 166, 212);
  font-size: larger;
}

.sign-header button {
  color: rgb(65, 78, 172);
  border-bottom: 3px solid rgb(160, 166, 212);
  font-size: larger;
  border-radius: 0;
}

.line {
  border-bottom: 0.3px solid #18CCCC;
  width: 98%;
}


form {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}


.signup-wrapper1 {
  width: 80%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  padding: 140px 0;
  border-radius: 10px;
  border: 2px solid #18CCCC;
  height: 90%;
  color: white !important;

}

.signup-wrapper2 {
  width: 80%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border-radius: 10px;
  border: 2px solid #18CCCC;
  height: 90%;
  color: white !important;
  /* margin: 10px 0; */

}

.signup-wrapper1 h1 {
  font-family: 'Roboto', sans-serif !important;
  font-size: xx-large;
  font-weight: bolder;
  color: white !important;
  margin: 0px;
  margin-top: 20px;
}

.signup-wrapper2 h1 {
  font-family: 'Roboto', sans-serif !important;
  font-size: xx-large;
  font-weight: bolder;
  color: white !important;
  margin: 0px;
  margin-top: 20px;

}

.sign-up-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  flex: 1;
}

.sign-up-form>input {
  width: 100%;
  height: 55px;
  border: 2px solid #ccc;
  border-radius: 5px;
}

.sign-up-form>input ::placeholder {
  color: white !important;
}

.sign-up-form-field label {
  color: white;

}

.sign-up-form-field {
  width: 100% !important;
  margin: 0px !important;
}

.otp-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  height: 20vh;
}

.otp-form>form>div {
  width: 80% !important;
}

.register-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  height: 100%;

}

.register-form form {
  width: 100%;
  height: 80%;
  margin: 20px 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.disabled-email {
  border: 1px solid rgb(74, 74, 74) !important;
  border-radius: 10px;
}

.disabled-email input {
  background-color: rgb(74, 74, 74) !important;
  color: rgb(255, 255, 255) !important;
}

.disabled-email>div>input>fieldset {
  background-color: rgb(74, 74, 74) !important;
  color: rgb(255, 255, 255) !important;
}

.disabled-email label {
  color: rgb(208, 208, 208) !important;

}

.register-form>form>div {
  width: 80% !important;
  margin-bottom: 20px;
}

.sign-up-button {
  font-family: 'Roboto', sans-serif !important;
  width: 100% !important;
  font-weight: 800 !important;
  color: white !important;
  background-color: #18CCCC !important;
  font-size: large !important;
}

.sign-up-form-field {
  width: 100%;


}

.button-wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  max-width: 30ch;
  width: 80%;
  height: 10%;
  color: white !important;
}

.button-wrapper h4 {
  margin-top: 20px;
  color: white !important;
  font-size: small !important;
}

.get-otp-button {
  font-family: 'Roboto', sans-serif !important;
  width: 100% !important;
  font-weight: 800 !important;
  color: white !important;
  background-color: #18CCCC !important;
  font-size: large !important;
}