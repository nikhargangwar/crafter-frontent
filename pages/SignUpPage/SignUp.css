@import "~react-image-gallery/styles/css/image-gallery.css";

.signup-page {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: row;
}

.signup-page-right {
  width: 40%;
}