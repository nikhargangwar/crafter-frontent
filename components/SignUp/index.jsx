/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import './SignUp.css';
import { FormValidation, makeRequest } from '../../utils';
import { TextField, Button, Snackbar, Box, Alert } from '@mui/material';
import { useNavigate } from 'react-router';
import { apiEndPoints } from '../../constants';
import icon from '../../assets/crafter-logo-dark.svg';
import { Link } from 'react-router-dom';
const SignUp = () => {
  const navigate = useNavigate();
  const [otpGenerated,setOtpGenerated]=useState(true);
  const [email,setEmail]=useState('');

  const [ registerData, setRegisterData ] = useState({
    name: '',
    otp:'',
    email: '',
    password: '',
    confirmPassword: ''
  });
  const [emailError,setEmailError]=useState(false);
  const [passwordError,setPasswordError]=useState(false);
  const [confirmPasswordError,setConfirmPasswordError]=useState(false);
  const [ snackbar, setSnackbar ] = useState({
    status: false,
    message: '',
    type: ''
  });

  const handleCloseSnackbar = () => {
    setSnackbar({ ...snackbar, status: false });
  };

  const handleOtpSubmit = async () => {
    console.log('otp submit');
    console.log(email);
    try {
      if (!FormValidation.validateEmail(email)) {
        setEmailError(true);
        return;
      }
      setSnackbar({ status: true, message: 'OTP Sent Successfully', severity: 'success' });

      const response = await makeRequest.authMakeRequest(apiEndPoints.generateOtp(email));
      setOtpGenerated(false);
      setEmail(email);

    } catch (error) {
      if(error.response) {
        setSnackbar({ status: true, message: error.response.data.message, severity: 'warning' });
        return;
      }
      setSnackbar({ status: true, message: 'Something went wrong! Please try again.', severity: 'warning' });
    }
  };



  const handleSubmit = async () => {
    try {
    
      if (!FormValidation.validatePassword(registerData.password)) {
        setPasswordError(true);
        return;
      }
    
      if (registerData.password !== registerData.confirmPassword) {
        setConfirmPasswordError(true);
        return;
      }
  
      const response = await makeRequest.authMakeRequest(apiEndPoints.registerUser(registerData.name, registerData.otp,registerData.email, registerData.password));
      setSnackbar({ status: true, message: response.message, severity: 'success' });
      setTimeout(() => {
        navigate('/login');
      }, 2000);
    } catch (error) {
      if(error.response) {
        setSnackbar({ status: true, message: error.response.data.message, severity: 'warning' });
        return;
      }
      setSnackbar({ status: true, message: 'Something went wrong! Please try again.', severity: 'warning' });
    }
  };

  return (
    <div className="sign-up">
      <div className="sign-up-form">
        <div className={otpGenerated ? 'signup-wrapper1' : 'signup-wrapper2'}>
          <img src={icon} onClick={() => navigate('/')} alt="icon" />
          <h1>SIGN UP</h1>
          <div className="sign-up-form-container">
            {
              otpGenerated? (<div className="otp-form">
                <Box
                  component="form"
                  sx={{
                    '& .MuiTextField-root': { m: 1,width: '30ch' },
                  }}
                  noValidate
                  autoComplete="off"
                >
                 
                  <div>
                    <TextField
                      error={emailError}
                      id="outlined-error"
                      label="Email"
                      value={email}
                      className="sign-up-form-field"
                      helperText={emailError? 'Enter a Valid Email' : ''}
                      sx={{ input: { color: 'white' ,borderRadius:'8px',helperText:{color:'white'}},fieldset:{border:'1px solid #18CCCC',color:'white '} } }
                      onChange={(e) => { setEmailError(false); setEmail(e.target.value); setRegisterData({ ...registerData, email: e.target.value  });}}
                      onKeyPress={(event) => { event.key === 'Enter' && handleOtpSubmit(); }}
                    />
                  </div>

                </Box>
                <div className="button-wrapper">
                  <Button variant='contained' className="get-otp-button" onClick={handleOtpSubmit} data-testid="get-otp-button">GET OTP</Button>
                  <h4>Already have an account? <Link to={'/login'} style={{color:'#18CCCC'}} on> Log in</Link></h4>
                </div>
                  
              </div>):(  <div className="register-form">
                <Box
                  component="form"
                  sx={{
                    '& .MuiTextField-root': { m: 1,width: '30ch' },
                  }}
                  noValidate
                  autoComplete="off"
                >
                  <div>
                    <TextField
                      
                      error={emailError}
                      id="outlined-error"
                      label="Email"
                      value={email}
                      sx={{ input: { color: 'white' ,borderRadius:'8px',helperText:{color:'white'},cursor:'not-allowed'},fieldset:{border:'1px solid white'} } }
                      className="sign-up-form-field disabled-email"
                      helperText={emailError? 'Enter a Valid Email' : ''}
                      onChange={(e) => { setEmailError(false); setRegisterData({ ...registerData, email: email }); }}
                      onKeyPress={(event) => { event.key === 'Enter' && handleSubmit(); }}
                    />
                  </div>
                  <div>
                    <TextField
                      id="outlined-error"
                      label="OTP"
                      value={registerData.otp}
                      className="sign-up-form-field"
                      sx={{ input: { color: 'white' ,borderRadius:'8px',helperText:{color:'white'}},fieldset:{border:'1px solid #18CCCC'} } }
                      onChange={(e) => { setRegisterData({ ...registerData, otp: e.target.value }); }}
                      onKeyPress={(event) => { event.key === 'Enter' && handleSubmit(); }}
                    />
                  </div>
                  <div>
                    <TextField
                      id="outlined-error"
                      label="Name"
                      value={registerData.name}
                      className="sign-up-form-field"
                      sx={{ input: { color: 'white' ,borderRadius:'8px',helperText:{color:'white'}},fieldset:{border:'1px solid #18CCCC'} } }
                      onChange={(e) => { setRegisterData({ ...registerData, name: e.target.value }); }}
                      onKeyPress={(event) => { event.key === 'Enter' && handleSubmit(); }}
                    />
                  </div>
                
                  <div>
                    <TextField
                      error={passwordError}
                      id="outlined-error"
                      label="Password"
                      value={registerData.password}
                      type='password'
                      sx={{ input: { color: 'white' ,borderRadius:'8px',helperText:{color:'white'}},fieldset:{border:'1px solid #18CCCC'} } }
                      className="sign-up-form-field"
                      helperText={passwordError? 'The password must have at least 8 characters long, contains at least one uppercase letter, one lowercase letter and one number' :''}
                      onChange={(e) => { setPasswordError(false); setConfirmPasswordError(false); setRegisterData({ ...registerData, password: e.target.value }); }}
                      onKeyPress={(event) => { event.key === 'Enter' && handleSubmit(); }}
                    />
                  </div>
                  <div>
                    <TextField
                      error={confirmPasswordError}
                      id="outlined-error"
                      label="Confirm Passwords"
                      type='password'
                      sx={{ input: { color: 'white' ,borderRadius:'8px',helperText:{color:'white'}},fieldset:{border:'1px solid #18CCCC'} } }
                      className="sign-up-form-field"
                      value={registerData.confirmPassword}
                      helperText={confirmPasswordError? 'Password and Confirm password must be the same' :''}
                      onChange={(e) => { setPasswordError(false); setConfirmPasswordError(false); setRegisterData({ ...registerData, confirmPassword: e.target.value }); }}
                      onKeyPress={(event) => { event.key === 'Enter' && handleSubmit(); }}
                    />
                  </div>
           
                </Box>
                <div className="button-wrapper">
                  <Button disabled={ registerData.name.length === 0 || passwordError || confirmPasswordError } variant='contained' className="sign-up-button" onClick={handleSubmit} data-testid="sign-up-button">SIGN UP</Button>
                  <h4>Already have an account? <Link to={'/login'} style={{color:'#18CCCC'}} on> Log in</Link></h4>
                </div>
              </div>)
            }
          
          </div>
          <Snackbar 
            open={snackbar.status}
            autoHideDuration={2000} 
            onClose={handleCloseSnackbar}
          >
            <Alert open={snackbar.status} onClose={handleCloseSnackbar} severity={snackbar.severity} sx={{ width: '100%' }}>
              {snackbar.message}
            </Alert>
          </Snackbar>  
        </div>
      </div>
     
      
    </div>
  );
};

export default SignUp;