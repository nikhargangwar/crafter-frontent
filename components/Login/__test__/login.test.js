import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import SignUp from '../index.js';
describe('Sign up', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<BrowserRouter><SignUp/></BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});