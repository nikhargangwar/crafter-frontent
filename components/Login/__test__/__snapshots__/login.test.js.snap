// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up should render correctly 1`] = `
<DocumentFragment>
  <div
    class="login"
  >
    <div
      class="login-form"
    >
      <div
        class="login-form-wrapper"
      >
        <img
          alt="icon"
          src="crafter-logo-dark.svg"
        />
        <p>
          LOGIN
        </p>
        <div
          class="login-form-fields"
        >
          <div
            class="MuiFormControl-root MuiTextField-root login-form-field css-bsxzbo-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r0:"
              id=":r0:-label"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root login-form-field css-bsxzbo-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r1:"
              id=":r1:-label"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="login-form-button-wrapper"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium login-form-button css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            LOGIN
          </button>
          <h4>
            Donâ€™t have an account? 
            <a
              href="/register"
              style="color: rgb(24, 204, 204); cursor: pointer;"
            >
               Create one now
            </a>
          </h4>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
